metabase/metabase
# Collection of UI updates to the alert modal

Closes https://github.com/metabase/metabase/issues/48428
Closes https://github.com/metabase/metabase/issues/48407
Closes https://github.com/metabase/metabase/issues/48402

### Description
A collection of small fixes to the Alert modal

### How to verify

1. Set up an instance with Email and a couple webhooks configured
2. Go to a question, and start creating an alert
3. Notice that Slack isn't displayed. Now disable Email and enable your webhooks
4. Create the alert
5. Go back to the alert modal and click "Delete this alert"
6. the names of the webhooks will be rendered in the checklist, and "Email" isn't in the checklist

Bonus:
1. Have an instance with Email and Slack set up
2. Go to a question and start creating a modal
3. Enable Slack. Notice that the "Done" button is greyed out, because you haven't yet selected a channel (previously, you could have still created the alert).


### Checklist
